launcher.py

```diff
@@ -0,0 +1,358 @@
+            os.remove(installer_name)
+
+
+def check_MediaProcessor(system):
+    if system == "Windows":
+        MediaProcessor_path = Path("MediaProcessor") / "build" / "MediaProcessor.exe"
+    else:
+        MediaProcessor_path = Path("MediaProcessor") / "build" / "MediaProcessor"
+    return MediaProcessor_path.exists()
+
+
+def create_virtualenv():
+    print(f"Creating virtual environment at: {venv_dir}")
+    execute_command(f"{sys.executable} -m venv {str(venv_dir)} --system-site-packages")
+    print("Virtual environment created successfully.")
+
+
+def install_python_dependencies(system):
+    try:
+        if not venv_dir.exists():
+            create_virtualenv()
+
+        print("Installing Python dependencies...")
+
+        if system == "Windows":
+            # Check for both `Scripts` and `bin` folders.
+            pip_path = venv_dir / "Scripts" / "pip.exe"
+            if not pip_path.exists():
+                pip_path = venv_dir / "bin" / "pip.exe"
+            if not pip_path.exists():
+                raise FileNotFoundError("pip not found in either Scripts or bin folder.")
+        else:
+            pip_path = venv_dir / "bin" / "pip"
+            if not pip_path.exists():
+                raise FileNotFoundError("pip not found in the bin folder.")
+
+        execute_command(f"{str(pip_path)} install -r requirements.txt")
+        print("Python dependencies installed.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to install Python dependencies: {e}")
+        sys.exit(1)
+
+
+def build_cpp_dependencies(system):
+    try:
+        print("Building MediaProcessor...")
+        build_dir = os.path.join("MediaProcessor", "build")
+        if not os.path.exists(build_dir):
+            os.makedirs(build_dir)
+
+        execute_command("cmake -DCMAKE_BUILD_TYPE=Release ..", cwd=build_dir)
+        execute_command("cmake --build . --config Release", cwd=build_dir)
+
+        print("MediaProcessor built successfully.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to build MediaProcessor: {e}")
+        sys.exit(1)
+
+
+def update_config(system):
+    try:
+        with open(CONFIG_FILE, "r") as config_file:
+            config = json.load(config_file)
+
+        if system == "Windows":
+            for key, value in config.items():
+                if type(value) == str:
+                    config[key] = value.replace("/", "\\")
+            config["ffmpeg_path"] = "ffmpeg"
+        elif system == "Darwin":
+            config["ffmpeg_path"] = "/usr/local/bin/ffmpeg"
+        elif system == "Linux":
+            config["ffmpeg_path"] = "/usr/bin/ffmpeg"
+
+        with open(CONFIG_FILE, "w") as config_file:
+            json.dump(config, config_file, indent=4)
+
+    except FileNotFoundError:
+        print(f"{CONFIG_FILE} not found. Please create a default config.json file.")
+        sys.exit(1)
+
+
+def launch_web_application(system):
```

Comment:
Would be good to see the logs and even perhaps start it with debug level. Currently if the process fails there is no way to see what happened when started via the launcher.